docs/bsp.md

```diff
@@ -1207,6 +1208,92 @@ Clients can use this information to reconstruct the tree of sbt meta builds. The
 `parent` information can be defined from `children` but it's provided by the server to
 simplify the data processing on the client side.
 
+## Connecting to a BSP Server
+
+The Build Server Protocol defines a standard to connect to BSP servers.
+This standard has been designed such that:
+
+1. Clients don't need to have specific support for a server to be able to
+   connect to it.
+1. Clients can connect to build tools installed both at the machine or at the
+   project level.
+1. Different versions of the same build tool can have different connection
+   mechanisms.
+
+The following protocol to discover and connect to BSP servers is inspired by
+[Jupyter's
+Kernelspec](https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernelspecs)
+and incorporates changes required to deal with  requirements above.
+
+### The BSP Connection Details
+
+The BSP connection details are modelled in a JSON object generated by a build
+tool command. The object contains the following fields to allow clients to
+connect to the server.
+
+```scala
+trait BspConnectionDetails {
+  /** The name of the build tool. */
+  def name: String
+  /** The version of the build tool. */
+  def version: String
+  /** A collection of languages supported by this BSP server. */
+  def languages: List[String]
+  /** A sequence of commands to connect to the BSP server that
+    * can be executed via system processes. */
+  def argv: List[String]
+}
+```
+
+Every build tool supporting BSP must implement a build-tool-specific command to
+generate the BSP connection details. The `argv` field in the connection details
+must be runnable via system process and can use whichever connection protocol
+is desired (UNIX Sockets, Windows Named Pipes, TCP, et cetera). The BSP server
```

Comment:
Must the server and client communicate via stdin/stdout or can they use a custom connection protocol? The paragraph says both 

- "whichever connection protocol" and
- "will write messages to `stdout` and will receive messages via `stdin`"

Reply:
This is an error on my side, I wrote the wrong thing, I'll fix it.